Kernel/Process.cpp

```diff
@@ -2486,8 +2486,9 @@ KResultOr<siginfo_t> Process::do_waitid(idtype_t idtype, int id, int options)
         return KResult(-EINVAL);
     }
 
-    if (Thread::current->block<Thread::WaitBlocker>(options, waitee_pid) != Thread::BlockResult::WokeNormally)
-        return KResult(-EINTR);
+    if (!(options & WNOHANG))
+        if (Thread::current->block<Thread::WaitBlocker>(options, waitee_pid) != Thread::BlockResult::WokeNormally)
+            return KResult(-EINTR);
```

Comment:
Seems like we should handle this in the `if (options & WNOHANG)` block above this code, no?
Also please wrap multi-line blocks in `{ }` :)

Reply:
we should not-block the thread?

I could assign that to a variable, I guess?

Reply:
Hmm hold on.. This code is organized a bit weirdly.

The above block will return from the function if `WNOHANG` *except* if the waitee process is not dead. It looks like a happy accident that we fall through for stopped children and then go through the same checks again.

Can we organize this code so it reads a bit more clearly and you can easily tell what's going on? :)

Reply:
Actually, come think of it, why is it even structured like this?
maybe we should just let `Thread::block()` happen, but exit immediately in WaitBlocker if WNOHANG?
(I'm trying to wrap my head around what would happen if we request P_ALL and WNOHANG)

Reply:
That sounds like it could be a lot more elegant yeah. Great idea :)